gremlin-dotnet/src/Gremlin.Net/Driver/ConnectionPool.cs

```diff
@@ -74,6 +85,36 @@ private void AddConnection(Connection connection)
             }
         }
 
+        private void ConsiderUnavailable()
+        {
+            CloseAndRemoveAllConnections();
+        }
+
+        private void CloseAndRemoveAllConnections()
+        {
+            lock (_connectionsLock)
+            {
+                TeardownAsync().WaitUnwrap();
+                RemoveAllConnections();
+            }
+        }
+
+        private void RemoveAllConnections()
+        {
+            while (!_connections.IsEmpty)
```

Comment:
Using `TryTake()` and `IsEmpty` can create race conditions.
Use `while (bag.TryTake(out c))` instead.